brave/brave-core
# Brave VPN card widget in NTP

Resolves https://github.com/brave/brave-browser/issues/41778

![Screenshot 2024-11-04 130047](https://github.com/user-attachments/assets/f566ad31-c55a-462a-9814-d10acd082b84)

https://github.com/user-attachments/assets/1ca8a484-3090-4b83-ab90-085539141435

<img src="https://github.com/user-attachments/assets/89f12514-19b9-40fa-8dd0-f6f19fdce463" alt="vpn purchased" width="300"/>
<img src="https://github.com/user-attachments/assets/0ba353f8-b618-4aea-98f8-646ea2e346e9" alt="vpn purchased" width="300"/>
<img src="https://github.com/user-attachments/assets/ddb71d57-f795-4d31-811d-edd06322b850" alt="vpn purchased" width="300"/>
<img src="https://github.com/user-attachments/assets/87c3df97-39fa-4818-b202-e5b7712e4604" alt="vpn purchased" width="300"/>
<img src="https://github.com/user-attachments/assets/3f387533-b86a-42c9-aa47-21f33ff91beb" alt="vpn purchased" width="300"/>

TODOs:
- [x] https://github.com/brave/brave-browser/issues/41973
- [x] https://github.com/brave/brave-browser/issues/41991
- [x] Polish UI
- [x] Fix build failure on Linux/Android 



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Launch with clean profile
2. Brave VPN card should be top
3. Launch with existing profile
4. Brave VPN card should be top
5. Click promotion card's `Free trial` button and check account page is loaded
6. Load vpn credentials from account page
7. Check card shows main page (toggle button)
8. Turn on/off with that toggle button and check vpn is connected/disconnected
9. Click `Change` link and check vpn panel is opened with server selection
10. Change to another region and check vpn card also has same state
